packages/driver/src/dom/visibility.js

```diff
@@ -22,56 +9,124 @@ const fixedOrAbsoluteRe = /(fixed|absolute)/
+  if ($elements.isOption(el) || $elements.isOptgroup(el)) {
+    // they could have just set to hide the option
+    if (elHasDisplayNone($el)) {
+      return true
+    }
+
+    // if its parent select is visible, then it's not hidden
+    const $select = $elements.getFirstParentWithTagName($el, 'select')
+
+    // check $select.length here first
+    // they may have not put the option into a select el,
+    // in which case it will fall through to regular visibility logic
+    if ($select && $select.length) {
+      // if the select is hidden, the options in it are visible too
+      return isHidden($select[0], name)
     }
+  }
 
-    throw new Error(`Cypress.dom.${name} must be passed a basic DOM element.`)
+  // anything with opacity 0 is not visible to eye
+  // TODO: comment back in on a breaking version change for visibility fixes
+  // if (elHasEffectiveOpacityZero($el)) {
+  //   return true // is hidden
+  // }
+
+  // in Cypress-land we consider the element hidden if
+  // either its offsetHeight or offsetWidth is 0 because
+  // it is impossible for the user to interact with this element
+  if (elHasNoEffectiveWidthOrHeight($el)) {
+    return true // is hidden
   }
 
-  const $el = $jquery.wrap(el)
+  // additionally if the effective visibility of the element
+  // is hidden (which includes any parent nodes) then the user
+  // cannot interact with this element and thus it is hidden
+  if (elHasVisibilityHiddenOrCollapse($el)) {
+    return true // is hidden
+  }
+
+  // we do some calculations taking into account the parents
+  // to see if its hidden by a parent
+  if (elIsHiddenByAncestors($el)) {
+    return true // is hidden
+  }
+
+  if (elOrAncestorIsFixed($el)) {
+    return elIsNotElementFromPoint($el)
+  }
 
-  //# in Cypress-land we consider the element hidden if
-  //# either its offsetHeight or offsetWidth is 0 because
-  //# it is impossible for the user to interact with this element
-  //# offsetHeight / offsetWidth includes the ef
-  return elHasNoEffectiveWidthOrHeight($el) ||
-
-    //# additionally if the effective visibility of the element
-    //# is hidden (which includes any parent nodes) then the user
-    //# cannot interact with this element and thus it is hidden
-    elHasVisibilityHidden($el) ||
-
-      //# we do some calculations taking into account the parents
-      //# to see if its hidden by a parent
-      elIsHiddenByAncestors($el) ||
-
-        //# if this is a fixed element check if its covered
-        (
-          elIsFixed($el) ?
-            elIsNotElementFromPoint($el)
-            :
-            //# else check if el is outside the bounds
-            //# of its ancestors overflow
-            elIsOutOfBoundsOfAncestorsOverflow($el)
-        )
+  // else check if el is outside the bounds
+  // of its ancestors overflow
+  return elIsOutOfBoundsOfAncestorsOverflow($el)
 }
 
+// const elHasEffectiveOpacityZero = ($el) => {
```

Comment:
Probably because I'm being lazy about resolving git conflicts later. 😬 😬 	

I plan to have these checks with our larger visibility changes, so just commented it out. 😬